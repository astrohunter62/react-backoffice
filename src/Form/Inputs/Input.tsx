import React, { FunctionComponent } from "react";
import { TextField, InputAdornment, MenuItem } from "@material-ui/core";
import { TYPES } from "../constants";

type InputEndProps = {
  icon?: JSX.Element;
};

const InputEnd: FunctionComponent<InputEndProps> = ({ icon }) => {
  return icon ? <InputAdornment position="end">{icon}</InputAdornment> : null;
};

type Props = {
  id: string;
  type?: string;
  title?: string;
  defaultValue?: string | any[] | number;
  rows?: number;
  isMultiline?: boolean;
  helperText?: string;
  isRequired?: boolean;
  error?: boolean;
  className?: string;
  select?: boolean;
  options?: string[];
  isDisabled?: boolean;
  iconEnd?: JSX.Element;
  onFocus?: (...args: any[]) => any;
  onBlur?: (...args: any[]) => any;
  onKeyPress?: (...args: any[]) => any;
  onChange: (event: any) => any;
  value?: any;
};

const Input: FunctionComponent<Props> = ({
  id,
  type = TYPES.TEXT,
  title,
  isMultiline = false,
  helperText,
  defaultValue,
  select = false,
  options = [],
  rows = 1,
  isRequired = false,
  error = false,
  iconEnd,
  isDisabled = false,
  className,
  onFocus,
  onBlur,
  onKeyPress,
  onChange,
  value,
}) => {
  return (
    <TextField
      id={id.replace(/[\[,\]]/g, "-")}
      type={type}
      label={title}
      value={type === "number" && Number.isNaN(value) ? "" : value}
      multiline={isMultiline}
      onChange={onChange}
      helperText={helperText}
      defaultValue={defaultValue}
      select={select}
      required={isRequired}
      error={error}
      margin="normal"
      fullWidth
      rows={rows}
      disabled={isDisabled}
      onFocus={onFocus}
      onBlur={onBlur}
      onKeyPress={onKeyPress}
      InputProps={{
        endAdornment: <InputEnd icon={iconEnd} />,
      }}
      className={className}
    >
      {options &&
        options.map((option) => (
          <MenuItem value={option} key={option.toLowerCase().replace(/ /, "")}>
            {option}
          </MenuItem>
        ))}
    </TextField>
  );
};

export default Input;
