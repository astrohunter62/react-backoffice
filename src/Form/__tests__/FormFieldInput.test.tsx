import React from "react";
import { shallow } from "enzyme";

import Input from "../Inputs/Input";

it("renders correctly", () => {
  const tree = shallow(<Input id="" onChange={() => {}} />);

  expect(tree).toMatchSnapshot();
});

it("renders disabled", () => {
  const tree = shallow(<Input id="" onChange={() => {}} isDisabled />);

  expect(tree).toMatchSnapshot();
});
