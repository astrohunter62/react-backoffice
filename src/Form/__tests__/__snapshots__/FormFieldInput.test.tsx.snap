// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<WithStyles(ForwardRef(TextField))
  InputProps={
    Object {
      "endAdornment": <InputEnd />,
    }
  }
  disabled={false}
  error={false}
  fullWidth={true}
  id=""
  margin="normal"
  multiline={false}
  required={false}
  rows={1}
  select={false}
  type="text"
/>
`;

exports[`renders disabled 1`] = `
<WithStyles(ForwardRef(TextField))
  InputProps={
    Object {
      "endAdornment": <InputEnd />,
    }
  }
  disabled={true}
  error={false}
  fullWidth={true}
  id=""
  margin="normal"
  multiline={false}
  required={false}
  rows={1}
  select={false}
  type="text"
/>
`;
